activerecord/lib/active_record/relation/record_fetch_warning.rb

```diff
@@ -34,14 +32,14 @@ def exec_queries
       class QueryRegistry # :nodoc:
         extend ActiveSupport::PerThreadRegistry
 
-        attr_accessor :queries
+        attr_reader :queries
 
         def initialize
-          reset
+          @queries = []
         end
 
         def reset
-          @queries = []
+          @queries.clear
```

Comment:
How often is this getting run? I don't think this is worth it for one object per request.